anvil/src/lib.rs

```diff
@@ -162,15 +162,19 @@ pub async fn spawn(mut config: NodeConfig) -> (EthApi, NodeHandle) {
     let node_service =
         tokio::task::spawn(NodeService::new(pool, backend, miner, fee_history_service, filters));
 
-    let host = config.host.unwrap_or(IpAddr::V4(Ipv4Addr::LOCALHOST));
-    let mut addr = SocketAddr::new(host, port);
+    let mut servers = Vec::new();
+    let mut addresses = Vec::new();
 
-    // configure the rpc server and use its actual local address
-    let server = server::serve(addr, api.clone(), server_config);
-    addr = server.local_addr();
+    for addr in config.host.iter() {
```

Comment:
if host is empty this should still use localhost

Reply:
Is there any valid use case where host can be an empty Vec? IMO it would be better to ensure that the values in NodeConfig are correct before the actual use. This will help keeping the code more readable. For this reason I added the default values in NodeConfig and NodeArgs. If you like this approach it can be further developed(in different pull request) by separating instantiation and validation in another NodeConfigBuilder struct.

What do you think? Of course I'll add the check if you prefer it that way :slightly_smiling_face:

Reply:
@mattsse are you OK to do the check in the NodeConfig::with_host method ?
```diff
--- a/anvil/src/config.rs
+++ b/anvil/src/config.rs
@@ -713,7 +713,7 @@ impl NodeConfig {
     /// Sets the host the server will listen on
     #[must_use]
     pub fn with_host(mut self, host: Vec<IpAddr>) -> Self {
-        self.host = host;
+        self.host = if host.is_empty() { vec![IpAddr::V4(Ipv4Addr::LOCALHOST)] } else { host };
         self
     }
```

or you prefer do it here in spawn ?

Reply:
that makes sense, yes

looks good